microsetta_private_api/api/_interested_user.py

```diff
@@ -95,15 +96,46 @@ def get_interested_user_address_update(interested_user_id, email):
                     message="Invalid user."
                 ), 404
             else:
-                # if the address is already valid, no reason to update
+                # If the address is already valid, no reason to update.
                 if interested_user.address_valid:
                     return jsonify(
                         code=400,
                         message="Address already verified."
                     ), 400
                 else:
-                    # we've determined it's a valid user and their address
-                    # needs to be fixed, so we return a subset of their info
+                    # We've determined it's a valid user and their address
+                    # needs to be fixed, so we return a subset of their info.
+                    # Certain paths into the database allow null address_2
+                    # which causes problems, so we change it to "".
+                    if interested_user.address_2 is None:
+                        interested_user.address_2 = ""
+
+                    # We also need to grab the reason(s) that Melissa couldn't
+                    # verify the address.
+                    melissa_repo = MelissaRepo(t)
+                    melissa_result = melissa_repo.check_duplicate(
+                        interested_user.address_1,
+                        interested_user.address_2,
+                        interested_user.address_3,
+                        interested_user.postal_code,
+                        interested_user.country
+                    )
+
+                    if isinstance(melissa_result, bool):
```

Comment:
```suggestion
                    if melissa_result is False:
```

The docstring and method only define `False` and at the moment it doesn't seem like there is a way to return `True`